### PR TITLE
enhance(lookup): tab autocomplete in all lookup controls

### PR DESCRIPTION
**Pull request by jonathanyeung**
## enhance(lookup): tab autocomplete in all lookup controls

This change adds tab autocomplete to all lookup controls, as well as in the web extension. 

Previously, if a command wanted tab autocomplete, it needed to implement an `onAutoComplete` interface, and then register itself with a singleton registrar during init. Any places that wanted to run the command would need to talk to the registrar and retrieve the right command instance out of it.

Instead, the mechanism now uses the `vscode.event` / `eventEmitter` pattern.  If you want to get notified of when the 'tab' key gets hit, you just get an `onAutocomplete` event out of the registrar (still a singleton for now), and then register /unregister your own callback in your own class. This simplifies the flow a lot and makes it much easier to add tab autocomplete functionality anywhere.

Related: https://github.com/dendronhq/dendron/issues/2393

### CODE REVIEW

**Comment by kevinslin** [comment]
can we add tests for this?

File: packages/plugin-core/src/web/extension.ts [hunk 1]
```diff
@@ -52,6 +53,21 @@ async function setupCommands(context: vscode.ExtensionContext) {
...
+    NoteLookupAutoCompleteCommand
+  );
+  const tabAutoCompletekey = DENDRON_COMMANDS.LOOKUP_NOTE_AUTO_COMPLETE.key;
+
+  if (!existingCommands.includes(tabAutoCompletekey))
+    context.subscriptions.push(
+      vscode.commands.registerCommand(
+        tabAutoCompletekey,
+        async (_args: any) => {
+          await TabAutoCompleteCommand.run();
```
Comment (kevinslin) [inline packages/plugin-core/src/web/extension.ts:66]: we can drop the await here and just return this, no?

**Review by kevinslin — COMMENTED** [review]
nice! big improvement in dx as well as functionality

**Comment by jonathanyeung** [comment]
tests added

**Review by kevinslin — APPROVED** [review]

**Pull request merged**